### PR TITLE
Creating a data model

### PR DESCRIPTION
**Pull request by melashu**
# In this milestone I done the following changes 

- I have created four table migrations.
- I have created foreign key migrations.
- Configuring the project with the PostgreSQL database.

### CODE REVIEW

File: db/migrate/20221207084440_add_user_to_likes.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUserToLikes < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :user, null: false, foreign_key: true
```
Comment (DammyShittu) [inline db/migrate/20221207084440_add_user_to_likes.rb:3]: - [ ] The like table should have an `author_id`, not a `user_id`. That means this reference should be `author`, not `user`. Kindly change that. You can point rails to the user table using the `to_table` option for the foreign key attribute.

  See this answer: [https://stackoverflow.com/a/42056089](https://stackoverflow.com/a/42056089)

File: db/migrate/20221207084950_add_user_ref_to_comments.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToComments < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :user, null: false, foreign_key: true
```
Comment (DammyShittu) [inline db/migrate/20221207084950_add_user_ref_to_comments.rb:3]: - [ ] The comments table should have an `author_id`, not a `user_id`. That means this reference should be `author`, not `user`. Kindly change that. You can point rails to the user table using the `to_table` option for the foreign key attribute.

  See this answer: [https://stackoverflow.com/a/42056089](https://stackoverflow.com/a/42056089)

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @melashu,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @melashu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**